libespm/include/libespm/CTDA.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+#include <cstdint>
+
+namespace espm {
+
+// Not a record type, but a field type commonly used across different record
+// types
+struct CTDA
+{
+  enum class Operator : uint8_t
+  {
+    EqualTo = 0,
+    NotEqualTo = 1,
+    GreaterThen = 2,
+    GreaterThenOrEqualTo = 3,
+    LessThen = 4,
+    LessThenOrEqualTo = 5
+  };
+
+  enum class Flags : uint8_t
+  {
+    ANDORDEFAULT = 0x00,
+    OR = 0x01,
+    Parameters = 0x02,
+    UseGlobal = 0x04,
+    UsePackData = 0x08,
+    SwapSubject = 0x10
+  };
+
+  enum class RunOnTypeFlags : uint32_t
+  {
+    Subject = 0,
+    Target = 1,
+    Reference = 2,
+    CombatTarget = 3,
+    LinkedReference = 4,
+    QuestAlias = 5,
+    PackageData = 6,
+    EventData = 7
+  };
+
+  struct DefaultData
+  {
+    uint32_t firstParameter;
```

Comment:
Please assign zeros by default for all struct members everywhere

like uint32_t firstParameter = 0; and so on